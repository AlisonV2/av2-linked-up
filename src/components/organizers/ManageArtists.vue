<template>
  <div class="row">
    <div class="col-12" v-for="event in events" :key="event.id">
      <h2 class="title">{{ event.name }}</h2>
      <p>Artists attending:</p>
      <div v-for="artist in event.stands" :key="artist.userId">
        -
        <router-link :to="{ name: 'Profile', params: { id: artist.userId } }">{{
          artist.userName
        }}</router-link>
      </div>
    </div>
  </div>
</template>

<script>
/**
 * @exports ManageArtists
 * @type {Component}
 * @vue-data {array} events
 * @vue-event created - Dispatch store action
 */
export default {
  name: 'ManageArtists',
  data() {
    return {
      events: [],
    };
  },
  created() {
    this.$store.dispatch('getOrgaEvents').then(() => {
      this.events = this.$store.getters.getOrgaEvents;
    });
  },
};
</script>
