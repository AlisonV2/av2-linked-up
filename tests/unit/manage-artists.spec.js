import { shallowMount, RouterLinkStub } from '@vue/test-utils';
import { createStore } from 'vuex';
import ManageArtists from '@/components/organizers/ManageArtists';

const actions = {
  getOrgaEvents: jest.fn(),
};

const getters = {
  getOrgaEvents: jest.fn(),
};
const store = createStore({
  actions,
  getters,
});

const $route = {
  name: 'Profile',
  params: { id: '1' },
};

const config = {
  data() {
    return {
      events: [],
      attendees: null,
    };
  },
  global: {
    plugins: [store],
    stubs: {
      RouterLink: RouterLinkStub,
    },
    components: {
      ManageArtists,
    },
    mocks: {
      $route,
    },
  },
};

/**
 * @module ManageArtistsTest
 */
describe('ManageArtists.vue', () => {
  it('ManageArtists Snapshot', () => {
    const component = shallowMount(ManageArtists, config);
    expect(component.element).toMatchSnapshot();
  });

  it('getAllArtists action is called on created', () => {
    const component = shallowMount(ManageArtists, config);
    ManageArtists.created.call(component.vm);
    expect(actions.getOrgaEvents).toHaveBeenCalled();
  });

  it('getAllArtists getters is called on created', () => {
    const component = shallowMount(ManageArtists, config);
    ManageArtists.created.call(component.vm);
    expect(getters.getOrgaEvents).toHaveBeenCalled();
  });
});
