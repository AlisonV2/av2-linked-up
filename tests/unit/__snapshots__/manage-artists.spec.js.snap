// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ManageArtists.vue ManageArtists Snapshot 1`] = `
<div
  class="row"
>
  
  
</div>
`;
